test/include/UTransportMock.h

```diff
@@ -13,5 +13,98 @@
 #define UP_CPP_TEST_UTRANSPORTMOCK_H
 
 #include <gmock/gmock.h>
+#include <up-cpp/transport/UTransport.h>
+
+namespace uprotocol::test {
+
+struct UTransportMockInfo {
+    bool    init_passed;
+    bool    deinit_passed;
+    size_t  send_count;
+    bool    cleanup_passed;
+    uprotocol::v1::UCode send_status_code;
+    std::string send_status_message;
+	uprotocol::utils::callbacks::CallerHandle<void, uprotocol::v1::UMessage const&> listener;
+	std::optional<uprotocol::v1::UUri> source_filter;
+
+    std::string payload;
+
+    void reset()
+    {
+        init_passed = false;
+        deinit_passed = false;
+        send_count = 0;
+        cleanup_passed = false;
+        send_status_code = v1::UCode::OK;      
+    }
+
+    UTransportMockInfo() { reset(); }
+
+    void dump(std::ostream& os)
+    {
+        using namespace std;
+        os << "MockInfo ------------------------------" << endl;
+        os << "init_passed         = " << init_passed << endl;
+        os << "deinit_passed       = " << deinit_passed << endl;
+        os << "cleanup_passed      = " << cleanup_passed << endl;
+        os << "send_count          = " << send_count << endl;
+        os << "send_status_code    = " << send_status_code << endl;
+        os << "send_status_message = " << send_status_message << endl;
+    }
+
+    void mock_packet(const std::string& mock_payload)
+    {
+        using namespace std;
+
+		uprotocol::v1::UMessage msg;
+        auto attr = new uprotocol::v1::UAttributes();
+		msg.set_allocated_attributes(attr);
+        msg.set_payload(mock_payload);
+        listener(msg);
+    }
+};
+
+class UTransportMock : public uprotocol::transport::UTransport {
+
+public:
+    static UTransportMockInfo mock_info;
+
+    explicit UTransportMock(const v1::UUri& uuri) : uprotocol::transport::UTransport(uuri)
+    {
+        mock_info.init_passed = true;      
+    }
+
+	~UTransportMock()
+    {
+        mock_info.deinit_passed = true;     
+    }
+
+private:
+	[[nodiscard]] v1::UStatus sendImpl(const v1::UMessage& message) override {
+        v1::UStatus retval;
+        mock_info.payload = message.payload();
+        retval.set_code(mock_info.send_status_code);
+        retval.set_message(mock_info.send_status_message);
+        mock_info.send_count++;
+        return retval;
+	}
+
+	[[nodiscard]] v1::UStatus registerListenerImpl(
+	    const v1::UUri& sink_filter, CallableConn&& listener,
+	    std::optional<v1::UUri>&& source_filter) override {
+        mock_info.listener = listener;
+        mock_info.source_filter = source_filter;
```

Comment:
Also capture the sink_filter:
```suggestion
        mock_info.source_filter = source_filter;
        mock_info.sink_filter = sink_filter;
```

Reply:
I've added the sink_filter to the capture. Will add checking in test layer in future push.